parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -334,6 +334,13 @@ write_files:
   content: !!binary |
     {{GetVariableProperty "cloudInitData" "ensureIMDSRestrictionScript"}}
 
+- path: /opt/azure/containers/cloud-init-status-check.sh
```

Comment:
uhh, lets think about karpenter here.

there is not concept of placing files on the disk through custom data for karpenter. if our bootstrapping relies on this file, we are going to break karpenter with the next release.

are we making sure that this file is also present on disk and the contents are exactly the same? i.e we will never get in a case where the file contents being placed from cloud init will overwrite whats on the disk.

Reply:
My understanding is that since we are adding this file to the VHD as well, karpenter case won't be affected.

-  If they use a new VHD with old custom data -> file will still be there on the VHD. 
- If they get an old VHD but new cse/custom data -> file will be added with cloud-init

Wanted to handle this like that since it seemed like the "cleanest" way to get proper logging in `cse_cmd.sh`